src/omero_rois/library.py

```diff
@@ -134,3 +134,36 @@ def masks_from_label_image(
                                       raise_on_no_mask)
         masks.append(mask)
     return masks
+
+
+def masks_from_3d_label_image(
+      planes, rgba=None, z=None, c=None, z_stack=True, t=None, text=None):
+    """
+    Create mask shapes from a 3d label image, either z stack or timepoints,
+    grouped by label ID (background=0)
+
+    :param list of numpy.array planes: list of 2D label arrays in z order
+    :param rgba int-4-tuple: Optional (red, green, blue, alpha) colour
+    :param z: Optional Z-index for the mask
+    :param c: Optional C-index for the mask
+    :param t: Optional T-index for the mask
+    :param z_stack: Flag if the planes represent a z stack, timepoints
+                    otherwise (default: True)
+    :param text: Optional text for the mask
+    :return: A dictionary of OMERO masks with the labels as keys
+           ({} if no labels found)
+
+    """
+    masks = {}
+    for i, plane in enumerate(planes):
+        if z_stack:
+            plane_masks = masks_from_label_image(plane, rgba, i, c, t,
+                                                 text, False)
+        else:
+            plane_masks = masks_from_label_image(plane, rgba, z, c, i,
+                                                 text, False)
+        for label, mask in enumerate(plane_masks):
+            if label not in masks:
+                masks[label] = []
```

Comment:
Also note `defaultdict` from `collections`.

Reply:
Isn't that code going to empty an existing `masks[label]` when the label is missing from one of the planes independently of its content?

For future unit test, this might be a scenario we need to test anyways e.g. the first plane containing labels (0,1,2) while the second plane only contains labels (0,1)

Reply:
Hm, not really, or I do miss something!? If the masks dictionary doesn't have an entry for a particular label yet, I have to add an empty list to it, so that I can append the masks for that label to it in the further steps.

Reply:
But I can use the `defaultdic` too, if that's clearer. Thanks @mtbc .

Reply:
Ignore my comment. I misread the code.

Reply:
Np. I pushed another commit using `defaultdic` like Mark suggested.